src/components/Sidebar/SidebarButton.tsx

```diff
@@ -19,7 +21,7 @@ const SidebarButton: FC<SidebarButtonProps> = ({ buttonName, buttonIcon, buttonT
         className={styles.sidebarButton}
         variant="contained"
         color="primary"
-        startIcon={buttonIcon}
+        startIcon={<FontAwesomeIcon icon={buttonIcon} />}
```

Comment:
Does this break the never pass around JSX as a prop rule?

Reply:
Need to look into this. In short yes it does but `Button` is a material component which requires a `ReactNode` as the `startIcon`. You can see the type of `buttonIcon` changed. It used to be a reactNode but now it is just the `IconDefinition`. Since we need to pass in the `ReactNode` to this material component there is nothing we can do. However, we can assume the material components are all optimized so we don't need to worry about it. The important part is that the `SidebarButton` component no longer takes in a `ReactNode` as a prop